components/authz-service/server/v2/policy.go

```diff
@@ -48,9 +50,10 @@ func NewMemstorePolicyServer(
 	l logger.Logger,
 	e engine.V2pXWriter,
 	pl storage_v1.PoliciesLister,
+	vSwitch *VersionSwitch,
 	vChan chan api.Version) (PolicyServer, PolicyRefresher, error) {
 
-	return NewPoliciesServer(ctx, l, memstore.New(), e, pl, vChan)
+	return NewPoliciesServer(ctx, l, memstore.New(), e, pl, vSwitch, vChan)
```

Comment:
Of course, vSwitch and vChan are related but they serve different functions, so I thought it best to keep them cleanly separate.

Reply:
`vSwitch` is two things:
- an UnaryServerInterceptor, denying v1 reqs when using v2 and vice-versa (this PreconditionFailed response drives the gateway IAM version switching)
- a struct keeping an api.Version

It would be better, I think, as outlined in #853, to have a version field in the server struct, and have that server implement the Interceptor.

`vChan` is a channel we're using to flip the version in that struct; based on a successful migration (upgrade-to-v2) *or* read on server startup (ie. when it's already been switched to IAM v2). It's a **shortcut**; it shouldn't be needed. We're creating a portal between two pieces of the service (the migration logic and the policies server), and have the former throw something through it at the latter. It's not how things should be, this is not something inherently calling for the use of a channel.